src/script.js

```diff
@@ -169,410 +159,19 @@ function isCanonicalSignature (buffer) {
-}
-
-// {signature} {pubKey}
-function pubKeyHashInput (signature, pubKey) {
-  typeforce(types.tuple(types.Buffer, types.Buffer), arguments)
-
-  return compile([signature, pubKey])
-}
-
-// <scriptSig> {serialized scriptPubKey script}
-function scriptHashInput (scriptSig, scriptPubKey) {
-  var scriptSigChunks = decompile(scriptSig)
-  var serializedScriptPubKey = compile(scriptPubKey)
-
-  return compile([].concat(
-    scriptSigChunks,
-    serializedScriptPubKey
-  ))
-}
-
-// <scriptSig> {serialized scriptPubKey script}
-function witnessScriptHashInput (scriptSig, scriptPubKey) {
-  return scriptHashInput(scriptSig, scriptPubKey)
-}
-
-// OP_0 [signatures ...]
-function multisigInput (signatures, scriptPubKey) {
-  if (scriptPubKey) {
-    var scriptData = decodeMultisigOutput(scriptPubKey, decompile(scriptPubKey))
-    if (signatures.length < scriptData.nRequiredSigs) throw new Error('Not enough signatures provided')
-    if (signatures.length > scriptData.nPublicKeys) throw new Error('Too many signatures provided')
-  }
-
-  return compile([].concat(OPS.OP_0, signatures))
-}
-
-function nullDataOutput (data) {
-  return compile([OPS.OP_RETURN, data])
-}
-
 module.exports = {
   compile: compile,
   decompile: decompile,
   fromASM: fromASM,
   toASM: toASM,
-
   number: require('./script_number'),
 
   isCanonicalPubKey: isCanonicalPubKey,
   isCanonicalSignature: isCanonicalSignature,
-  isDefinedHashType: isDefinedHashType,
-
-  decodePubKeyOutput: decodePubKeyOutput,
-  decodePubKeyHashOutput: decodePubKeyHashOutput,
-  decodeMultisigOutput: decodeMultisigOutput,
-  decodeScriptHashOutput: decodeScriptHashOutput,
-  decodeNullDataOutput: decodeNullDataOutput,
-  decodeWitnessPubKeyHashOutput: decodeWitnessPubKeyHashOutput,
-  decodeWitnessScriptHashOutput: decodeWitnessScriptHashOutput,
-
-  isPubKeyOutput: isPubKeyOutput,
-  isPubKeyHashOutput: isPubKeyHashOutput,
-  isMultisigOutput: isMultisigOutput,
-  isScriptHashOutput: isScriptHashOutput,
-  isNullDataOutput: isNullDataOutput,
-  isWitnessPubKeyHashOutput: isWitnessPubKeyHashOutput,
-  isWitnessScriptHashOutput: isWitnessScriptHashOutput,
-  classifyOutput: classifyOutput,
-
-  isPubKeyInput: isPubKeyInput,
-  isPubKeyHashInput: isPubKeyHashInput,
-  isMultisigInput: isMultisigInput,
-  isScriptHashInput: isScriptHashInput,
-  classifyInput: classifyInput,
-
-  pubKeyOutput: pubKeyOutput,
-  pubKeyHashOutput: pubKeyHashOutput,
-  multisigOutput: multisigOutput,
-  scriptHashOutput: scriptHashOutput,
-  nullDataOutput: nullDataOutput,
-  witnessPubKeyHashOutput: witnessPubKeyHashOutput,
-  witnessScriptHashOutput: witnessScriptHashOutput,
-
-  pubKeyInput: pubKeyInput,
-  pubKeyHashInput: pubKeyHashInput,
-  multisigInput: multisigInput,
-  scriptHashInput: scriptHashInput,
-  witnessScriptHashInput: witnessScriptHashInput
+  isDefinedHashType: isDefinedHashType
+}
+
+var templates = require('./templates')
+for (var key in templates) {
+  module.exports[key] = templates[key]
```

Comment:
As mentioned, maybe this would be better as `bitcoin.templates`... though tbh,  `bitcoin.script.templates` makes more contextual sense.